lxd/storage_zfs.go

```diff
@@ -63,6 +63,13 @@ func (s *storageZfs) StorageCoreInit() error {
 	}
 	s.sTypeName = typeName
 
+	if !shared.RunningInUserNS() {
```

Comment:
I'm also not sure we want to fail if "loadModule" fails here. There could be cases where modprobe is confused but the module is available. So a fire and forget

    loadModule("zfs")

Where we ignore whatever it returns to us should be just fine. The call to the zfs tool right afterwards will catch everything else.